### ISSUE TITLE
Component Change Request: use AAA contrast standards on Error notification bar

### ISSUE THREAD

**Issue by maureenlholland** [mozilla/protocol#848]
### Please describe the change you wish to see

Aim for AAA WCAG color contrast standards on [Errors notification bars](https://protocol.mozilla.org/components/detail/notification-bar--error.html)

### Please describe why these changes need to be made

The current Error notification styles pass normal AA contrast checks but fail AAA normal text check
<img width="581" alt="current-contrast" src="https://user-images.githubusercontent.com/19650432/216604919-5fc97f36-1202-43b9-97cf-94fa51f63856.png">

In the case of Error notifications, it feels like we should go beyond normal requirements and make the text as easy to read as possible. Errors are already frustrating and it would compound user frustration to struggle to read the error description.
<img width="512" alt="error" src="https://user-images.githubusercontent.com/19650432/216605048-9cd4f771-763c-42a0-8e70-e5f225baf5f9.png">

### Please provide any designs or prototypes of the proposed changes

Enforcing large/bolded text might be the most effective solution. 

### Please describe where this component is currently in use

Screenshot above is from Newsletter error (locally forcing errors from the views.py file: http://localhost:8000/en-US/newsletter/confirm/thanks/?error=2)

### Please describe what this component should do in the following conditions

* When set in front of a dark background
* When set in front of a light background
* When viewed in a mobile/responsive viewport

I don't think it needs any changes to responsive viewport behaviour or dark/light theme

### Is the development of this component a blocking dependency for other work? Please explain if so

Not blocking

**Comment by reemhamz** [comment]
Thanks for creating this issue! It's easy to realize that even though color contrast passes, our eyes may just generally need something a little better than what the green `PASS` stamp tells us.

I fiddled around with the error notification bar using dev tools. Two things I found most helpful (conducted separately):
- Changing font color to plain old fashion black (#000000)
![image](https://user-images.githubusercontent.com/42309026/217147215-618fdf98-5395-4e6d-ac73-7ed1aa977ce7.png)

- Increasing the font size to `16px`

![image](https://user-images.githubusercontent.com/42309026/217147506-af841017-8a82-4b8a-8118-093bd0316881.png)

Both seem to improve the notification bar (just increasing size by `2px` made such a difference for me), and I would prefer to add both of the changes. However, if we were only to choose one method, I would lean towards increasing the font-size rather than the color. 

Also, I noticed that the rest of the notification bars seem to have a similar issue to what you're describing. Color contrast passes, but it's visually not-so-pleasant to read. I can see myself squinting if i wasn't sitting hunched and so close to my laptop screen 😆.

I'd be happy to make that change to both color and size for the rest of the notification bars. Happy to discuss this further.

**Comment by maureenlholland** [comment]
@reemhamz Thanks for your thoughts! I agree the font-size increase is more effective than the font color adjustment alone.

I initially thought large & bolded would be best, but we seem to reserve bold styling for [clickable things](https://github.com/mozilla/protocol/blob/main/assets/sass/protocol/components/_notification-bar.scss#L151) in the notification bars.

The success notification bar is in all the WNPs. We should check if a font size increase there is off-putting before changing all the component font-sizes. I think it would be acceptable for the error notification alone to have larger font size (similar to how the clickable notification is bolded, it's part of the specific intention of that type of notification to be a bit more in-your-face). Changing the font color to black across all notification component styles should be OK in any case.

**Comment by reemhamz** [comment]
@maureenlholland 

#### `14px` in WNP's notification bar
![image](https://user-images.githubusercontent.com/42309026/218662093-3e8a57cb-e581-430e-a708-f2bb5bc9eeae.png)

VS.

#### `16px` in WNP's notification bar
![image](https://user-images.githubusercontent.com/42309026/218662185-6e1715e1-c95f-465c-bf6d-59288290fb48.png)

I don't think the `16px` is too off-putting, per se. But a user would be more aware of the size opposed to the former `14px`. Though a slight change, I notice it significantly more. And the hierarchical structure of elements seems to look a tad unordered.

Safest thing I can think of doing is changing the font-color to black, as you mentioned above.

**Comment by maureenlholland** [comment]
PR merged, but leaving this issue open for further discussion (feel free to close and open a new issue specifically for standardizing error colors across Protocol components)

From [Craig's comment on the PR](https://github.com/mozilla/protocol/pull/850#issuecomment-1432083599): 

> Though another option would be to darken the background and invert the text color. We use white on a dark red for inline error messages in forms and white on a not-as-dark red for error lists. (https://protocol.mozilla.org/components/detail/error-messages.html and https://protocol.mozilla.org/components/detail/error-list.html). I think we could revisit both of those and perhaps land on one standard for error states across the board.

**Comment by reemhamz** [comment]
Made a quick test via devtools and checked out the color contrast using webAIM:

(**note**: color used was `$color-red-80` which is `#810220`)

![image](https://user-images.githubusercontent.com/42309026/220212788-2ee59901-e9bd-4d28-9928-9be920de06b2.png)


![image](https://user-images.githubusercontent.com/42309026/220212651-4f41cc94-bba7-44c5-aae1-4e51a5785ff6.png)

![Screen Shot 2023-02-21 at 10 28 03 AM](https://user-images.githubusercontent.com/42309026/220212599-bfdb0f0e-7a4f-452c-900c-73da903b4b7d.png)


It has the highest contrast ratio from all the reds we tested and imo wouldn't be a bad color to use for the rest of Protocol. However, looking at it from a color psychology POV, does darker red add more stress opposed to lighter reds? Would love to head what @ehot has to say.

**Comment by reemhamz** [comment]
cc: @maureenlholland @craigcook @ehot for visibility here.

**Comment by maureenlholland** [comment]
Thanks for the ping @reemhamz 

Going to preface this by saying I don't have a strong opinion beyond, let's make it AAA level contrast, but aesthetically I feel this dark red + white text combination is a tough fit with our other, more vibrant notification background colors.

That said, the error notification color combo (while passing AAA standards) might not be a great fit for small text form errors 
![Screenshot 2023-02-28 at 10 10 55 AM](https://user-images.githubusercontent.com/19650432/221824708-a2f0bf43-f766-4045-8372-be68a5f85345.png)

Maybe it's possible to have a primary error style (default) and secondary error style (when you need that extra bit of contrast)? I defer to you and Elise and Craig for the standardization decision.

**Comment by craigcook** [comment]
I think we can try a shade lighter red, as well as changing the text color to black on the form errors.

**Comment by reemhamz** [comment]
Will be attempting `$color-red-30` (#ff848b) for error messages for forms.
